src/kex_ntru/kex_ntru.c

```diff
@@ -116,10 +119,12 @@ int OQS_KEX_ntru_alice_0(UNUSED OQS_KEX *k, void **alice_priv, uint8_t **alice_m
 
 err:
 	ret = 0;
-	if (ntru_alice_priv != NULL)
+	if (ntru_alice_priv != NULL && ntru_alice_priv->priv_key != NULL)
```

Comment:
Below you wrote "All contemporary implementations of `free()` can be called with a NULL pointer. The majority of code in liboqs does not check to see if the pointer is NULL before calling `free()`.".  Does that also apply here for `ntru_alice_priv->priv_key`?

Reply:
Absolutely, good catch. That must have been a knee-jerk reaction to try and deal with `ntru_alice_priv->priv_key` not being initialized. The real problem is of course if `ntru_crypto_ntru_encrypt_keygen` fails we end up here with an uninitialized pointer in `ntru_alice_priv->priv_key`. I will update the code.